convert2rhel/utils.py

```diff
@@ -66,6 +69,141 @@ class Color(object):
 BACKUP_DIR = os.path.join(TMP_DIR, "backup")
 
 
+# Code taken from https://stackoverflow.com/a/63773140
+class Process(multiprocessing.Process):
+    """Overrides the implementation of the multiprocessing.Process class."""
+
+    def __init__(self, *args, **kwargs):
+        """Default constructor for the class"""
+        multiprocessing.Process.__init__(self, *args, **kwargs)
+        self._pconn, self._cconn = multiprocessing.Pipe()
+        self._exception = None
+
+    def run(self):
+        """Overrides the `run` method to catch exceptions raised in child."""
+        try:
+            multiprocessing.Process.run(self)
+            self._cconn.send(None)
+        # Here, `SystemExit` is inherit from `BaseException`, which is too
+        # broad to catch as it involves for-loop exceptions. The ideal here
+        # is to catch `SystemExit` *and* any `Exception` that shows up as we do
+        # a lot of logger.critical() and they do raise `SystemExit`.
+        except (Exception, SystemExit) as e:
+            tb = traceback.format_exc()
+            self._cconn.send((e, tb))
+
+    @property
+    def exception(self):
+        """Custom property to access the exception raised by the children."""
+        if self._pconn.poll():
+            self._exception = self._pconn.recv()
+        return self._exception
```

Comment:
Not sure if it is necessary to do unit tests for this class. Let me know what you all think about.